sync-core/src/main/java/com/cloudant/sync/query/IndexUpdater.java

```diff
@@ -318,6 +318,12 @@ private DBParameter populateDBParameter(List<String> fieldNames,
             Object value = ValueExtractor.extractValueForFieldName(fieldName, rev.getBody());
 
             if (value != null) {
+                // Do not include an empty array field as you would not include a
+                // field with no value.
+                if (value instanceof List && ((List) value).size() == 0) {
+                    continue;
+                }
+
                 includeFieldNames.add(fieldName);
                 args.add(value);
```

Comment:
There is a check above that skips already included fields which means that it skips already included arguments `initialArgs`.  Take a look a few lines up https://github.com/cloudant/sync-android/pull/167/files#diff-0d8c3940dfc1c0e1639ac3ad7ed9a2c5R314.